Jenkinsfile

```diff
@@ -30,18 +30,72 @@ pipeline {
 		stage('Setup: Back up DB'){
 			steps{
 				script{
-          withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
-            def release_current_before_biomodels_dump = "${env.RELEASE_CURRENT}_${currentRelease}_before_biomodels.dump"
-            sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_before_biomodels_dump"
-            sh "gzip -f $release_current_before_biomodels_dump"
-          }
+					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+						def release_current_before_biomodels_dump = "${env.RELEASE_CURRENT}_${currentRelease}_before_biomodels.dump"
+						sh "mysqldump -u$user -p$pass ${env.RELEASE_CURRENT} > $release_current_before_biomodels_dump"
+						sh "gzip -f $release_current_before_biomodels_dump"
+          				}
+				}
+			}
+		}
+		stage('Setup: Create reactome database from release_current') {
+			steps{
+				script{
+					withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]) {
+						sh "mysql -u$user -p$pass -e \'drop database if exists ${env.REACTOME}; create database ${env.REACTOME}\'"
+						sh "zcat ${env.RELEASE_CURRENT}_${currentRelease}_before_biomodels.dump.gz | mysql -u$user -p$pass ${env.REACTOME}"
+					}
+				}
+			}
+		}
+		// This stage generates the graph database using the graph-importer module, and replaces the current graph db with it.
+		stage('Setup: Generate Graph Database'){
+			steps{
+				script{
+					cloneOrPullGitRepo("release-jenkins-utils")
+					sh "cp -f release-jenkins-utils/scripts/changeGraphDatabase.sh ${env.JENKINS_HOME_PATH}"
+					sh "chmod 700 ${env.JENKINS_HOME_PATH}changeGraphDatabase.sh"
+					cloneOrPullGitRepo("graph-importer")
+					dir("graph-importer"){
+						sh "mvn clean compile assembly:single"
+						withCredentials([usernamePassword(credentialsId: 'mySQLUsernamePassword', passwordVariable: 'pass', usernameVariable: 'user')]){
+							sh "java -jar target/GraphImporter-jar-with-dependencies.jar --name ${env.REACTOME} --user $user --password $pass --neo4j ./graph.db --interactions"
+							sh "tar -zcf graphdb_${currentRelease}_biomodels.tgz graph.db/"
+							sh "mv graph.db /tmp/"
+							sh "sudo service tomcat7 stop"
+							sh "sudo service neo4j stop"
+							// This static script adjusts permissions of the graph.db folder and moves it to /var/lib/neo4j/data/databases/.
+							sh "sudo bash ${env.JENKINS_HOME_PATH}changeGraphDatabase.sh"
+							sh "sudo service neo4j start"
+							sh "sudo service tomcat7 start"
+							sh "rm ${env.JENKINS_HOME_PATH}changeGraphDatabase.sh"
```

Comment:
This is not the only place where the graph database gets generated. Is it possible to move this part of the Step to some sort of common library so that multiple steps can use the same code for generating the graph database?

Reply:
Yea I realize that would be better, it could be consolidated into a commonly used graph-database-importer jenkinsfile in release-common-utils, I just wanted to write everything out since this is the first release that this is being done. If we want to leave this PR open until I do that, thats fine, though I'd like to use this for this release since I know it works

Reply:
No that's OK. I mostly wanted to just raise it as a discussion point. You could create an issue to track progress on that task, but this is not a blocker for now.

Reply:
I agree with @SolomonShorser-OICR - creating an issue for tracking would be good but no need to block this PR.